### ISSUE TITLE
Make `generateUsername` function deterministic

### ISSUE THREAD

**Issue by tomast1337** [OpenNBS/NoteBlockWorld#14]
The current implementation of the [generateUsername](https://github.com/OpenNBS/NoteBlockWorld/blob/7c79a7005bcfa5da7efe0f15dcaec8aefd6c9dda/server/src/user/user.service.ts#L101) function relies on a random number to generate unique usernames when the base username is already taken. This can lead to non-deterministic behavior and potentially "non-terminating" states.

Proposed Solution
Modify the [generateUsername](https://github.com/OpenNBS/NoteBlockWorld/blob/7c79a7005bcfa5da7efe0f15dcaec8aefd6c9dda/server/src/user/user.service.ts#L101) function to use a counter instead of a random number. This will ensure deterministic and unique username generation.

```ts
public async generateUsername(inputUsername: string) {
  // Normalize username (remove accents, replace spaces with underscores)
  const baseUsername = inputUsername
    .replace(' ', '_')
    .normalize('NFKD')
    .replace(/[\u0300-\u036f]/g, '')
    .replace(/[^a-zA-Z0-9_]/g, '');
  let newUsername = baseUsername;

  // Find if there's already a user with the same username
  let nameTaken = await this.usernameExists(baseUsername);
  let counter = 1;

  while (nameTaken) {
    newUsername = `${baseUsername}_${counter}`;
    nameTaken = await this.usernameExists(newUsername);
    counter++;
  }

  return newUsername;
}
```

While this solution addresses the issue, it's more a quick fix rather than a comprehensive one.

**Comment by Bentroen** [comment]
Looks good to me!

**Comment by tomast1337** [comment]
Merged fix with #15 to develop
